File: Android-Firebase-Notification-SourceCode/app/src/main/java/com/hellohasan/android_firebase_notification/activity/MessageShowActivity.java
package com.hellohasan.android_firebase_notification.activity;

import androidx.appcompat.app.AppCompatActivity;
import android.os.Bundle;
import android.widget.ImageView;
import android.widget.TextView;

import com.hellohasan.android_firebase_notification.R;
import com.squareup.picasso.Picasso;

public class MessageShowActivity extends AppCompatActivity {

    private ImageView imageView;
    private TextView titleTextView;
    private TextView timeStampTextView;
    private TextView articleTextView;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_message_show);
        viewInitialization();

        //receive data from MyFirebaseMessagingService class
        String title = getIntent().getStringExtra("title");
        String timeStamp = getIntent().getStringExtra("timestamp");
        String article = getIntent().getStringExtra("article_data");
        String imageUrl = getIntent().getStringExtra("image");

        //Set data on UI
        titleTextView.setText(title);
        timeStampTextView.setText(timeStamp);
        articleTextView.setText(article);
        Picasso.get()
                .load(imageUrl)
                .error(R.drawable.default_image)
                .into(imageView);
    }

    private void viewInitialization() {
        imageView = findViewById(R.id.featureGraphics);
        titleTextView = findViewById(R.id.header);
        timeStampTextView = findViewById(R.id.timeStamp);
        articleTextView = findViewById(R.id.article);
    }
}

File: AndroidFirebaseNotification/app/src/main/java/com/hellohasan/android_firebase_notification/activity/MessageShowActivity.kt
package com.hellohasan.android_firebase_notification.activity

import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.view.MenuItem
import androidx.core.app.NavUtils

import com.hellohasan.android_firebase_notification.R
import com.squareup.picasso.Picasso
import kotlinx.android.synthetic.main.activity_message_show.*


class MessageShowActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_message_show)
        supportActionBar?.setDisplayHomeAsUpEnabled(true)

        //receive data from MyFirebaseMessagingService class
        val title = intent.getStringExtra("title")
        val timeStampString = intent.getStringExtra("timestamp")
        val articleString = intent.getStringExtra("article_data")
        val imageUrl = intent.getStringExtra("image")

        //Set data on UI
        Picasso.get()
            .load(imageUrl)
            .placeholder(R.drawable.image_placeholder)
            .error(R.drawable.image_placeholder)
            .into(featureGraphics)

        header.text = title
        timeStamp.text = timeStampString
        article.text = articleString
    }

    override fun onOptionsItemSelected(item: MenuItem?): Boolean {

        if (item?.itemId == android.R.id.home)
            onBackPressed()

        return super.onOptionsItemSelected(item)
    }

    override fun onBackPressed() {
        super.onBackPressed()

        val intent = NavUtils.getParentActivityIntent(this)
        startActivity(intent)
    }
}
